src/clj/com/rpl/specter.cljc

```diff
@@ -980,6 +980,28 @@
         (setval (srange index index) [v] structure)
         ))))
 
+(defrichnav
+  ^{:doc "Navigates to the empty space between the index and the prior index. For select
+          navigates to NONE, and transforms to non-NONE insert at that position."}
+  before-index-new
+  [index]
+  (select* [this vals structure next-fn]
+           NONE)
+  (transform* [this vals structure next-fn]
+              (let [v (next-fn vals NONE)]
+                (if (identical? NONE v)
+                  structure
+                  (let [updated (concat (take index structure) [v] (drop index structure))]
```

Comment:
Looks like this changes the type of the data structure to a sequence. Needs to be the same type as when it started.

Reply:
Should be fixed in the latest implementation